### PR TITLE
Add tests for bootstrap.py

### PR DESCRIPTION
**Pull request by gauthampughazhendhi**
Add tests for bootstrap.py

### CODE REVIEW

**Review by zackt113 — APPROVED** [review]
Looks good to me! Thanks for starting off a good sample! Will follow your structure for my function as well.

File: tests/test_bootstrap.py [hunk 1]
```diff
@@ -1 +1,106 @@
-from straPy import bootstrap
+import numpy as np
+from pytest import raises
+from straPy.bootstrap import bootstrap_distribution
+
+
+def test_bootstrap_distribution():
+    """
+    Tests the functionality of `bootstrap_distribution()`, which includes
+    checking the type of object returned, it's shape and it's values.
```
Comment (BooleanJulien) [inline tests/test_bootstrap.py:9]:
```suggestion
    checking the type of object returned, its shape and its values.
```

Comment (zackt113) [inline tests/test_bootstrap.py]: Haha, good catch

File: tests/test_bootstrap.py [hunk 2]
```diff
@@ -1 +1,106 @@
...
+    # checks the shape of returned object
+    assert dist.shape == (3,)
+
+    # checks the values of returned object
+    assert np.array_equal(dist.tolist(),
+                          [1.3333333333333333,
+                           1.6666666666666667,
+                           1.3333333333333333])
+
+    dist = bootstrap_distribution([1, 2, 3],
```
Comment (BooleanJulien) [inline tests/test_bootstrap.py]:
```suggestion
    dist_var_1 = bootstrap_distribution([1, 2, 3],
```

File: tests/test_bootstrap.py (hunk 1 above)
Comment (BooleanJulien) [inline tests/test_bootstrap.py]:
```suggestion
    checking the type of object returned, its shape and its values.
```

File: tests/test_bootstrap.py [hunk 3]
```diff
@@ -1 +1,106 @@
...
+                                  estimator="var",
+                                  random_seed=1)
+
+    # checks with different estimator
+    assert np.array_equal(dist.tolist(),
+                          [0.2222222222222222,
+                           0.2222222222222222,
+                           0.2222222222222222])
+
+    dist_2 = bootstrap_distribution([1, 2, 3],
```
Comment (BooleanJulien) [inline tests/test_bootstrap.py]:
```suggestion
    dist_var_2 = bootstrap_distribution([1, 2, 3],
```

File: tests/test_bootstrap.py [hunk 4]
```diff
@@ -1 +1,106 @@
...
+                                  3,
+                                  estimator="var",
+                                  random_seed=1)
+
+    # checks if random_seed works
+    assert np.array_equal(dist, dist_2)
```
Comment (BooleanJulien) [inline tests/test_bootstrap.py]:
```suggestion
    assert np.array_equal(dist_var_1, dist_var_2)
```

File: tests/test_bootstrap.py (hunk 2 above)
Comment (BooleanJulien) [reply tests/test_bootstrap.py]: I got confused in the later test because dist was reassigned, renaming for clarity

**Review by BooleanJulien — APPROVED** [review]
Looks great! Some very minor changes. Echoing Zack's comments that I will try my best to replicate this format. 

Thanks Gautham!

File: tests/test_bootstrap.py (hunk 2 above)
Comment (gauthampughazhendhi) [reply tests/test_bootstrap.py]: @BooleanJulien added the change. I reused the variable since the test requires only two variables.

**Pull request merged**